### ISSUE TITLE
[no-this-alias] incorrect report for destructuring

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#759]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-this-alias": ["error"]
  }
}
```

```TS
class Foo {
  prop: string = '';
  meth() {
    const {prop} = this;
  }
}
```

**Expected Result**
no reports

**Actual Result**
reports on the desctructuring

**Comment by ankeetmaini** [comment]
This is because `defaultOptions.allowDestructuring` is `false`. This works fine if we pass {allowDestructuring: true}

**Comment by bradzacher** [comment]
ahhhhhhh I fell victim to the classic blunder of not reading the docs. Such a fool!

I don't know why this wouldn't be the default behaviour.
I'll change this as part of the recommended config changes
